geomstats/backend/tensorflow/__init__.py

```diff
@@ -40,25 +40,33 @@ def get_mask_i_float(i, n):
     return mask_i_float
 
 
-def get_mask_float(indices, mask_shape):
+def get_mask_float(indices, mask_shape, dtype=float32):
     """Create a binary mask.
 
     Parameters
     ----------
-    indices : tuple
+    indices: tuple
         Single index or tuple of indices where ones will be.
-    mask_shape : tuple
+    mask_shape: tuple
         Shape of the mask.
+    dtype: dtype
+        Type of the mask.
 
     Returns
     -------
-    tf_mask : array
+    tf_mask : array, shape=[mask_shape]
     """
     np_mask = _np.zeros(mask_shape)
-    if ndim(array(indices)) <= 1 and ndim(np_mask) == 1:
-        indices = list(indices)
+    single_index = not isinstance(indices, list)
+
+    if single_index:
+        indices = [indices]
```

Comment:
The edge case single_index is not tested in the unit tests.

Reply:
True, the way it is coded, single_index is tested in `assignment_single_value`, which always call `get_mask_float` (or the vectorized version) with a list, even of length one. I'll correct it.

Reply:
Or should it be kept in case get_mask_float is directly called ? But with an additional test ?

Reply:
If `get_mask_float` is always called with a list, then maybe we can just remove this `if`?